### ISSUE TITLE
P2P connections still occupy server traffic

### ISSUE THREAD

**Issue by AlexFugui** [ntop/n2n#1090]
I built the server using n2n v3.0.0

Playing UDP multicast games with a few friends using Edge

Our nc command checks that everyone and others are connected in a peer-to-peer manner

However, checking server traffic still consumes a significant amount of bandwidth

In theory, P2P connections should no longer use server forwarding, and I don't know how to troubleshoot this issue

Even if only 2 people are playing online games, ensuring that a P2P link is used, server bandwidth is still occupied

**Comment by Logan007** [comment]
> In theory, P2P connections should no longer use server forwarding,

Yes. That's correct. But those edges who cannot connect p2p do use forwarding through server. However, broadcast/multicast traffic always goes through the server because edges do not necessarily know about whole network and all nodes connected.

>  and I don't know how to troubleshoot this issue

You could try leveraging p2p connections by various means. Maybe your router allows for port forwarding, be it a fixed port (`-p` on the the edge) or upnp/pmp assisted port forwarding (`tools/n2n-portfwd`).

**Comment by AlexFugui** [comment]
> > In theory, P2P connections should no longer use server forwarding,
> 
> Yes. That's correct. But those edges who cannot connect p2p do use direct link. However, broadcast/multicast traffic always goes through the server because edges do not necessarily know about whole network and all nodes connected.
> 
> > and I don't know how to troubleshoot this issue
> 
> You could try leveraging p2p connections by various means. Maybe your router allows for port forwarding, be it a fixed port (`-p` on the the edge) or upnp/pmp assisted port forwarding (`tools/n2n-portfwd`).

First of all, thank you very much
Thank you for your guidance
I will test the solution you provided on my own and friends' computers
I will reply to you with a clear conclusion in about two days

**Comment by AlexFugui** [comment]
> > In theory, P2P connections should no longer use server forwarding,
> 
> Yes. That's correct. But those edges who cannot connect p2p do use forwarding through server. However, broadcast/multicast traffic always goes through the server because edges do not necessarily know about whole network and all nodes connected.
> 
> > and I don't know how to troubleshoot this issue
> 
> You could try leveraging p2p connections by various means. Maybe your router allows for port forwarding, be it a fixed port (`-p` on the the edge) or upnp/pmp assisted port forwarding (`tools/n2n-portfwd`).

Thank you for your patience. I have only recently completed the test
Indeed, all multicast will be forwarded through the server
Using - p in edge cannot be avoided either
Perhaps I should look for ways to compress data to reduce server bandwidth pressure

**Comment by qq2656739379** [comment]
没有开upnp?
